### PR TITLE
🧹 refactor config gathering for sbom cmd

### PR DESCRIPTION
**Pull request by chris-rock**
This PR harmonizes the config gathering. Since sbom subcommand does not support all flags, the flag parsing has been converted to viper so that it can be used even if a specific flag is not supported.

### CODE REVIEW

File: apps/cnquery/cmd/root.go [hunk 1]
```diff
@@ -196,3 +204,103 @@ func GenerateMarkdown(dir string) error {
 
 	return nil
 }
+
+func getCobraScanConfig(cmd *cobra.Command, runtime *providers.Runtime, cliRes *plugin.ParseCLIRes) (*scanConfig, error) {
```
Comment (imilchev) [inline apps/cnquery/cmd/root.go:208]: this function can return an error but it seems like we are doing `os.Exit` and `log.Fatal()` instead. I think it's better to return an error and let the caller of the function decide what to do with that error

Comment (chris-rock) [reply apps/cnquery/cmd/root.go:208]: +1 good call. I'll adjust that

Comment (chris-rock) [reply apps/cnquery/cmd/root.go:208]: Code updated.

File: apps/cnquery/cmd/root.go [hunk 2]
```diff
@@ -196,3 +204,97 @@ func GenerateMarkdown(dir string) error {
...
+	opts, err := config.Read()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to load config")
```
Comment (imilchev) [inline apps/cnquery/cmd/root.go]: missed 1 here

**Review by imilchev — APPROVED** [review]

File: apps/cnquery/cmd/root.go (hunk 2 above)
Comment (chris-rock) [reply apps/cnquery/cmd/root.go]: oh no, will adjust

**Pull request merged**